File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 10/27/2025, 7:10:19 PM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxsum=currsum=nums[0]
        for i in range(1,len(nums)):
            currsum=max(nums[i],currsum+nums[i])
            maxsum=max(maxsum,currsum)

        return maxsum

File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 10/27/2025, 7:10:20 PM
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ms=INT_MIN,cs;
        for(int i=0;i<nums.size();i++){
            cs+=nums[i];
            ms=max(ms,cs);
            if(cs<0) cs=0;
        }
        return ms;
    }
};